### ISSUE TITLE
why in train.py，botch loss_disc and loss_gen use "generator_TPRLS_loss"?

### ISSUE THREAD

**Issue by lliwons** [yl4579/HiFTNet#3]
why in train.py，botch loss_disc and loss_gen use "generator_TPRLS_loss"?
loss_disc_s += generator_TPRLS_loss(y_ds_hat_r, y_ds_hat_g)
loss_gen_s += generator_TPRLS_loss(y_ds_hat_r, y_ds_hat_g)

**Comment by yl4579** [comment]
I think you are totally right. This is a typo in the cleaned up code. I have fixed it.

**Comment by lliwons** [comment]
Thanks.

**Comment by hdmjdp** [comment]
> why in train.py，botch loss_disc and loss_gen use "generator_TPRLS_loss"? loss_disc_s += generator_TPRLS_loss(y_ds_hat_r, y_ds_hat_g) loss_gen_s += generator_TPRLS_loss(y_ds_hat_r, y_ds_hat_g)

because， the two losses are exactly the same. @lliwons @yl4579 , maybe you do not need fix it?

**Comment by hdmjdp** [comment]
@yl4579 Are the same?
def discriminator_TPRLS_loss(disc_real_outputs, disc_generated_outputs):
    loss = 0
    for dr, dg in zip(disc_real_outputs, disc_generated_outputs):
        tau = 0.04
        m_DG = torch.median((dr-dg))
        L_rel = torch.mean((((dr - dg) - m_DG)**2)[dr < dg + m_DG])
        loss += tau - F.relu(tau - L_rel)
    return loss

def generator_TPRLS_loss(disc_real_outputs, disc_generated_outputs):
    loss = 0
    for dg, dr in zip(disc_real_outputs, disc_generated_outputs):
        tau = 0.04
        m_DG = torch.median((dr-dg))
        L_rel = torch.mean((((dr - dg) - m_DG)**2)[dr < dg + m_DG])
        loss += tau - F.relu(tau - L_rel)
    return loss

**Comment by yl4579** [comment]
No, the real and generated inputs are switched for the generator loss.
